NDTensors/test/ITensors/TestITensorDMRG/dmrg.jl

```diff
@@ -0,0 +1,28 @@
+using Test
+using ITensors
+using NDTensors
+using Random
+
+Random.seed!(1234)
+
+function test_dmrg(elt, N::Integer, dev::Function)
+  sites = siteinds("S=1/2", N)
+
+  os = OpSum()
+  for j in 1:(N - 1)
+    os += "Sz", j, "Sz", j + 1
+    os += 0.5, "S+", j, "S-", j + 1
+    os += 0.5, "S-", j, "S+", j + 1
+  end
+
+  H = dev(MPO(elt, os, sites; linkdims=4))
+
+  psi0 = dev(randomMPS(elt, sites;))
+
+  nsweeps = 7
+  cutoff = [1e-3, 1e-13]
+  noise = [1e-12, 0]
+
+  energy, psi = dmrg(H, psi0; nsweeps, cutoff, noise, outputlevel=0)
+  @test energy ≈ get_ref_value(dev, N, elt)
```

Comment:
I think we should add an `rtol` here to make the test a bit more forgiving, it can be tricky comparing floating point numbers in tests. Something like `rtol=10 * eps(real(elt))`.

Reply:
That will probably be particularly important to add if we set a lower `maxdim` and `nsweeps` since it will become less accurate. I think we just want to make sure the energy is accurate to some number of digits, we don't need to test very high precision.